// @atlaskit/flag/dist/cjs/components/Flag/styledFlag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Description = exports.Expander = exports.Content = exports.DismissButton = exports.Title = exports.Icon = exports.Header = exports.default = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _theme2 = require("../../theme");

var getBoxShadow = function getBoxShadow(props) {
  var borderColor = (0, _theme2.flagBorderColor)(props);
  var shadowColor = (0, _theme2.flagShadowColor)(props);
  var border = borderColor && "0 0 1px ".concat(borderColor);
  var shadow = "0 20px 32px -8px ".concat(shadowColor);
  return [border, shadow].filter(function (p) {
    return p;
  }).join(',');
};

var _default = _styledComponents.default.div.withConfig({
  displayName: "styledFlag",
  componentId: "qhniyv-0"
})(["\n  background-color: ", ";\n  border-radius: ", "px;\n  box-sizing: border-box;\n  box-shadow: ", ";\n  color: ", ";\n  padding: ", "px;\n  transition: background-color 200ms;\n  width: 100%;\n  z-index: 600;\n\n  &:focus {\n    outline: none;\n    box-shadow: 0 0 0 2px ", ";\n  }\n"], _theme2.flagBackgroundColor, _theme.borderRadius, getBoxShadow, _theme2.flagTextColor, _theme.math.multiply(_theme.gridSize, 2), _theme2.flagFocusRingColor); // Header


exports.default = _default;

var Header = _styledComponents.default.div.withConfig({
  displayName: "styledFlag__Header",
  componentId: "qhniyv-1"
})(["\n  display: flex;\n  align-items: center;\n  height: ", "px;\n"], _theme.math.multiply(_theme.gridSize, 4));

exports.Header = Header;

var Icon = _styledComponents.default.div.withConfig({
  displayName: "styledFlag__Icon",
  componentId: "qhniyv-2"
})(["\n  flex: 0 0 auto;\n  width: ", "px;\n"], _theme.math.multiply(_theme.gridSize, 5));

exports.Icon = Icon;

var Title = _styledComponents.default.span.withConfig({
  displayName: "styledFlag__Title",
  componentId: "qhniyv-3"
})(["\n  color: ", ";\n  font-weight: 600;\n  flex: 1;\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n"], _theme2.flagTextColor);

exports.Title = Title;

var DismissButton = _styledComponents.default.button.withConfig({
  displayName: "styledFlag__DismissButton",
  componentId: "qhniyv-4"
})(["\n  appearance: none;\n  background: none;\n  border: none;\n  border-radius: ", "px;\n  color: ", ";\n  cursor: pointer;\n  flex: 0 0 auto;\n  margin-left: ", "px;\n  padding: 0;\n  white-space: nowrap;\n\n  &:focus {\n    outline: none;\n    box-shadow: 0 0 0 2px ", ";\n  }\n"], _theme.borderRadius, _theme2.flagTextColor, _theme.gridSize, _theme2.flagFocusRingColor); // Content


exports.DismissButton = DismissButton;

var Content = _styledComponents.default.div.withConfig({
  displayName: "styledFlag__Content",
  componentId: "qhniyv-5"
})(["\n  display: flex;\n  flex: 1 1 100%;\n  flex-direction: column;\n  justify-content: center;\n  min-width: 0;\n  padding: 0 0 0 ", "px;\n"], _theme.math.multiply(_theme.gridSize, 5));

exports.Content = Content;

var Expander = _styledComponents.default.div.withConfig({
  displayName: "styledFlag__Expander",
  componentId: "qhniyv-6"
})(["\n  max-height: ", "px;\n  opacity: ", ";\n  overflow: ", ";\n  transition: max-height 0.3s, opacity 0.3s;\n"], function (_ref) {
  var isExpanded = _ref.isExpanded;
  return isExpanded ? 150 : 0;
}, function (_ref2) {
  var isExpanded = _ref2.isExpanded;
  return isExpanded ? 1 : 0;
}, function (_ref3) {
  var isExpanded = _ref3.isExpanded;
  return isExpanded ? 'visible' : 'hidden';
});

exports.Expander = Expander;

var Description = _styledComponents.default.div.withConfig({
  displayName: "styledFlag__Description",
  componentId: "qhniyv-7"
})(["\n  color: ", ";\n  word-wrap: break-word;\n"], _theme2.flagTextColor);

exports.Description = Description;
