// @atlaskit/flag/dist/cjs/components/FlagGroup/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _reactTransitionGroup = require("react-transition-group");

var _portal = _interopRequireDefault(require("@atlaskit/portal"));

var _theme = require("@atlaskit/theme");

var _Wrapper = _interopRequireDefault(require("../../styled/Wrapper"));

var _styledFlagGroup = _interopRequireWildcard(require("./styledFlagGroup"));

var FlagGroup =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(FlagGroup, _Component);

  function FlagGroup() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, FlagGroup);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(FlagGroup)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "props", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "renderChildren", function () {
      var _this$props = _this.props,
          children = _this$props.children,
          onDismissed = _this$props.onDismissed;
      return _react.Children.map(children, function (flag, idx) {
        var isDismissAllowed = idx === 0;
        var id = flag.props.id;
        return _react.default.createElement(_reactTransitionGroup.Transition, {
          key: id,
          addEndListener: function addEndListener(node, done) {
            node.addEventListener('animationend', done);
          }
        }, function (transitionState) {
          return _react.default.createElement(_Wrapper.default, {
            transitionState: transitionState
          }, (0, _react.cloneElement)(flag, {
            onDismissed: onDismissed,
            isDismissAllowed: isDismissAllowed
          }));
        });
      });
    });
    return _this;
  }

  (0, _createClass2.default)(FlagGroup, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(_portal.default, {
        zIndex: _theme.layers.flag()
      }, _react.default.createElement(_styledFlagGroup.default, null, _react.default.createElement(_styledFlagGroup.SROnly, null, "Flag notifications"), _react.default.createElement(_styledFlagGroup.Inner, {
        component: "div"
      }, this.renderChildren())));
    }
  }]);
  return FlagGroup;
}(_react.Component);

exports.default = FlagGroup;
